pinot-server/src/main/java/com/linkedin/pinot/server/request/SimpleRequestHandler.java

```diff
@@ -71,11 +69,19 @@ public SimpleRequestHandler(QueryExecutor queryExecutor, ServerMetrics serverMet
     BrokerRequest brokerRequest = null;
     try {
       final InstanceRequest queryRequest = new InstanceRequest();
-      serDe.deserialize(queryRequest, byteArray);
+      if (! serDe.deserialize(queryRequest, byteArray) ) {
```

Comment:
the var names in this method are a little confusing (queryRequest vs instanceRequest). Could you fix that? thanks